### PR TITLE
Fix Windows support for automatic code reference pages

### LINKED ISSUE #417: Broken MkDocs code annotation
Navigate to:

https://mkdocstrings.github.io/recipes/#bind-pages-to-sections-themselves

Note the unrendered `#2` and `#3` (probably a copy-paste oversight):

Screenshot 2022-04-09 193523

### LINKED ISSUE #418: Automatic code reference pages recipe breaks on Windows
**Describe the bug**

The recipe: https://mkdocstrings.github.io/recipes/#bind-pages-to-sections-themselves

Works on *Nix systems, but seems to break on Windows.

**To Reproduce**

1. Checkout: https://github.com/johnthagen/python-blueprint (pre https://github.com/johnthagen/python-blueprint/commit/f8a396d878be9e971ef4bad1fc963db8654a5775)
2. Install `tox`
3. Run `tox -e docs`

```
INFO     -  Cleaning site directory
INFO     -  Building documentation to directory: python-blueprint\site
INFO     -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
              - reference\fact\__init__.md                                                                   
WARNING  -  A relative path to 'reference/fact☻95♥_init__.md' is included in the 'nav' configuration, which is not
            found in the documentation files                                                                      
WARNING  -  Documentation file 'reference\SUMMARY.md' contains a link to 'reference\fact☻95♥_init__.md' which is not
            found in the documentation files.   
```

**Expected behavior**

MkDocs project builds as expected, even on Windows.

**Information (please complete the following information):**
- OS: Windows 10
- `mkdocstrings` version: 0.18.0

```
mkdocs                     1.2.3
mkdocs-autorefs            0.3.1
mkdocs-gen-files           0.3.4
mkdocs-htmlproofer-plugin  0.8.0
mkdocs-literate-nav        0.4.1
mkdocs-material            8.2.1
mkdocs-material-extensions 1.0.3
mkdocstrings               0.18.0
mkdocstrings-python        0.6.3
mkdocstrings-python-legacy 0.2.2
```

### PR DESCRIPTION
**Pull request by johnthagen**
Closes #418 

Closes #417

### CODE REVIEW

File: docs/recipes.md [hunk 1]
```diff
@@ -179,7 +179,7 @@ for path in sorted(Path("src").rglob("*.py")):
     elif parts[-1] == "__main__":
         continue
 
-    nav[parts] = str(doc_path)  # (1)
+    nav[parts] = str(PurePosixPath(doc_path))  # (1)
```
Comment (oprypin) [inline docs/recipes.md:182]:
```suggestion
    nav[parts] = doc_path.as_posix()  # (1)
```

Please double-check that my suggestion actually works and then please apply this style instead

Comment (johnthagen) [reply docs/recipes.md:182]: @oprypin Nice suggestion! I didn't know about `as_posix()`. That is certainly a cleaner way to get the same outcome.

**Comment by johnthagen** [comment]
@oprypin @pawamoy I've confirmed that this simplified implementation works on Windows.

**Review by pawamoy — APPROVED** [review]
Great! Looking good :slightly_smiling_face:

**Comment by johnthagen** [comment]
@pawamoy Please Squash the commits when you merge using the GitHub UI.

Thanks!

**Comment by Archmonger** [comment]
Thanks for this one John!

**Pull request merged**